src/gui/core/event/dispatcher.cpp

```diff
@@ -198,7 +198,7 @@ bool dispatcher::execute_hotkey(const hotkey::HOTKEY_COMMAND id)
 		return false;
 	}
 
-	return itor->second(*this, id);
+	return itor->second(dynamic_cast<widget&>(*this), id);
```

Comment:
I wonder, do we even have dispatchers which aren't widgets? Maybe you can outright merge the `dispatcher` and `widget` classes. (Albeit it's understandable if it's outside the scope of this refactoring.)